### PR TITLE
Create Gyro abstraction class

### LINKED ISSUE #12: Gyro Subsystem

### PR DESCRIPTION
**Pull request by GrandMoff100**
Implements median filter for x,y, and z angle position values from gyro.

### CODE REVIEW

**Comment by GrandMoff100** [comment]
Not done yet.

**Review by royaltwrecks — DISMISSED** [review]
Don't edit Robot.java

**Review by royaltwrecks — DISMISSED** [review]

**Comment by royaltwrecks** [comment]
@GrandMoff100 have you tested? This is Jack

**Comment by GrandMoff100** [comment]
Not yet. Try it with Sherman and output the angle positions to smart dashboard while driving.

**Review by TonyBaston — COMMENTED** [review]
Without explicit testing, I don't want to merge, but it looks good.

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 1]
```diff
@@ -19,7 +18,7 @@ public class Drivetrain extends SubsystemBase {
     RelativeEncoder rightEncoder = frontRighttDrive.getEncoder();
 
     DifferentialDrive drive = new DifferentialDrive(frontLeftDrive, frontRighttDrive);
-
+    Gyro gyro = new Gyro();
```
Comment (IrishConman) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:22]: I'd use a different variable name for the gyro object

**Review by IrishConman — DISMISSED** [review]
Looks good for now, but I'm uncertain if this code has been tested yet. Change the name of the gyro object to something more readable & test, then I'll approve the PR

File: src/main/java/frc/robot/subsystems/Drivetrain.java (hunk 1 above)
Comment (royaltwrecks) [reply src/main/java/frc/robot/subsystems/Drivetrain.java:22]: Done

**Comment by royaltwrecks** [comment]
Code has been tested.

**Review by royaltwrecks — DISMISSED** [review]

**Review by royaltwrecks — APPROVED** [review]

**Review by ilikepegasi — APPROVED** [review]

**Pull request merged**